### PR TITLE
[SDK-4096] Update In-App Messaging for Unity

### PR DESCRIPTION
**Pull request by bryanlogan**
# Pull Request/Issue Resolution

#### Description of Change:
https://braze-docs-11ztaeg7o-braze.vercel.app/docs/developer_guide/platform_integration_guides/unity/in-app_messaging/#custom-action-listeners

Closes SDK-4096

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [X ] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [ ] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [ ] Check that all links work.
- [ ] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @josh-mccrowell-braze and @bre-fitzgerald as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [ ] If the documentation involves a 1) paid SKU, 2) a third party, 3) SMS, 4) AI, or 5) privacy, ensure that Maria Maldonado on the Legal team has signed off.
- [ ] Tag others as reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @josh-mccrowell-braze or @bre-fitzgerald for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs<br>Currents</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Intelligence<br>In-App Messages<br>Channels<br>FIX</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging Experience<br>Message Components<br>Email (Composition and Infrastructure)</td>
  </tr>
  <tr>
    <td>@rachel-feinberg</td>
    <td>Customer Lifecycle, Identity and Permissions<br>SMS<br>User Targeting<br>Reporting</td>
  </tr>
</table>
</details>

### CODE REVIEW

**Comment by bryanlogan** [comment]
https://braze-docs-11ztaeg7o-braze.vercel.app/docs/developer_guide/platform_integration_guides/unity/in-app_messaging/#custom-action-listeners

**Review by davidbielik — APPROVED** [review]

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 1]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
...
+}
+
+public void OnInAppMessageDismissed(IInAppMessage inAppMessage) {
+  // executed whenever an in-app message is dismissed without a click.
+}
+```
+
+[8]: https://github.com/braze-inc/braze-unity-sdk/blob/master/Assets/Plugins/Appboy/Models/InAppMessage/InAppMessageButton.cs
+[12]: https://github.com/braze-inc/braze-unity-sdk/blob/master/Assets/Plugins/Appboy/Models/InAppMessage/IInAppMessageImmersive.cs
+[13]: https://github.com/braze-inc/braze-unity-sdk/blob/master/Assets/Plugins/Appboy/Models/InAppMessage/IInAppMessage.cs
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:132]:
```suggestion
[8]: https://github.com/braze-inc/braze-unity-sdk/blob/18cb8ee89f1841c576eb954793edb6e06f9130b4/Assets/Plugins/Appboy/Models/InAppMessage/InAppMessageButton.cs
[12]: https://github.com/braze-inc/braze-unity-sdk/blob/18cb8ee89f1841c576eb954793edb6e06f9130b4/Assets/Plugins/Appboy/Models/InAppMessage/IInAppMessageImmersive.cs
[13]: https://github.com/braze-inc/braze-unity-sdk/blob/18cb8ee89f1841c576eb954793edb6e06f9130b4/Assets/Plugins/Appboy/Models/InAppMessage/IInAppMessage.cs
```

These docs are at a fairly high chance of going stale given it's a wrapper so GH stable links are probably better.

**Review by radixdev — APPROVED** [review]

**Review by vanessaland — APPROVED** [review]

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 2]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
 
 Use `LogButtonClicked(int buttonID)` on [`IInAppMessageImmersive`][12] to log button clicks. Note that buttons are represented as lists of[`InAppMessageButton`][8] instances, each of which contains a `ButtonID`.
 
-[8]: https://github.com/Appboy/appboy-unity-sdk/blob/master/Assets/Plugins/Appboy/Models/InAppMessage/InAppMessageButton.cs
-[12]: https://github.com/Appboy/appboy-unity-sdk/blob/master/Assets/Plugins/Appboy/Models/InAppMessage/IInAppMessageImmersive.cs
-[13]: https://github.com/Appboy/appboy-unity-sdk/blob/master/Assets/Plugins/Appboy/Models/InAppMessage/IInAppMessage.cs
+## Custom Action Listeners
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:95]:
```suggestion
## Custom action listeners
```

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 3]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
...
+
+If you require more control over how a user interacts with in-app messages, use a `BrazeInAppMessageListener` and assign it to `Appboy.AppboyBinding.inAppMessageListener`. For any delegates you don't want to use, you can simply leave them as null.
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:97]:
```suggestion
If you require more control over how a user interacts with in-app messages, use a `BrazeInAppMessageListener` and assign it to `Appboy.AppboyBinding.inAppMessageListener`. For any delegates you don't want to use, you can simply leave them as `null`.
```

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 4]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
...
+  BeforeInAppMessageDisplayed = BeforeInAppMessageDisplayed,
+  OnInAppMessageButtonClicked = OnInAppMessageButtonClicked,
+  OnInAppMessageClicked       = OnInAppMessageClicked,
+  OnInAppMessageHTMLClicked   = OnInAppMessageHTMLClicked,
+  OnInAppMessageDismissed     = OnInAppMessageDismissed,
+};
+Appboy.AppboyBinding.inAppMessageListener = listener;
+
+public void BeforeInAppMessageDisplayed(IInAppMessage inAppMessage) {
+  // executed before an in-app message is displayed.
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:110]:
```suggestion
  // Executed before an in-app message is displayed.
```

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 5]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
...
+}
+
+public void OnInAppMessageButtonClicked(IInAppMessage inAppMessage, InAppMessageButton inAppMessageButton) {
+  // executed whenever an in-app message button is clicked.
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:114]:
```suggestion
  // Executed whenever an in-app message button is clicked.
```

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 6]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
...
+}
+
+public void OnInAppMessageClicked(IInAppMessage inAppMessage) {
+  // executed whenever an in-app message is clicked.
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:118]:
```suggestion
  // Executed whenever an in-app message is clicked.
```

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 7]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
...
+}
+
+public void OnInAppMessageHTMLClicked(IInAppMessage inAppMessage, Uri uri) {
+  // executed whenever an HTML in-app message is clicked.
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:122]:
```suggestion
  // Executed whenever an HTML in-app message is clicked.
```

File: _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md [hunk 8]
```diff
@@ -92,6 +92,41 @@ Use `LogClicked()` and `LogImpression()` on [`IInAppMessage`][13] to log clicks
...
+}
+
+public void OnInAppMessageDismissed(IInAppMessage inAppMessage) {
+  // executed whenever an in-app message is dismissed without a click.
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/unity/in-app_messaging.md:126]:
```suggestion
  // Executed whenever an in-app message is dismissed without a click.
```

**Review by josh-mccrowell-braze — APPROVED** [review]

**Review by josh-mccrowell-braze — APPROVED** [review]

**Pull request merged**